code/modules/vehicles/vehicle_actions.dm

```diff
@@ -345,20 +347,27 @@
 	COOLDOWN_START(vim_mecha, sound_cooldown, VIM_SOUND_COOLDOWN)
 	vehicle_entered_target.visible_message(span_notice("[vehicle_entered_target] [sound_message]"))
 	playsound(vim_mecha, sound_path, 75)
+	SEND_SIGNAL(vim_mecha, signal_fired)
 
 /datum/action/vehicle/sealed/noise/chime
 	name = "Chime!"
 	desc = "Affirmative!"
 	button_icon_state = "vim_chime"
 	sound_path = 'sound/machines/chime.ogg'
 	sound_message = "chimes!"
+	signal_fired = COMSIG_VIM_CHIME_USED
 
 /datum/action/vehicle/sealed/noise/buzz
 	name = "Buzz."
 	desc = "Negative!"
 	button_icon_state = "vim_buzz"
 	sound_path = 'sound/machines/buzz-sigh.ogg'
 	sound_message = "buzzes."
+	signal_fired = COMSIG_VIM_BUZZ_USED
 
 /datum/action/vehicle/sealed/headlights/vim
 	button_icon_state = "vim_headlights"
+
+/datum/action/vehicle/sealed/headlights/vim/Trigger(trigger_flags)
+	. = ..()
+	SEND_SIGNAL(vehicle_entered_target, COMSIG_VIM_HEADLIGHTS_TOGGLED, vehicle_entered_target.headlights_toggle)
```

Comment:
why send this info over?

Reply:
while writing this code, i was afraid that state of the headlights might at some point not match the pin output in the circuit module due to some oversight, so i figured that i could prevent that entirely from happening by sending this info over just in case

Reply:
alright, good enough